mitmproxy/addons/mapeditor.py

```diff
@@ -0,0 +1,95 @@
+import typing
+
+from mitmproxy import exceptions
+from mitmproxy import flowfilter
+from mitmproxy import ctx
+
+
+def parse_mapeditor(mexpr):
+    """
+        Returns a (pattern, path_to_file) tuple
+
+        example:
+            ~u .*://example.com/script.js:MAP_TO:/etc/hostname
+            ^                            ^      ^            ^
+            |------filter_expression-----|--sep-|path_to_file|
+    """
+    parts = mexpr.split(":MAP_TO:")
+    if len(parts) != 2:
+        raise exceptions.OptionsError(
+            "Invalid map editor specifier: %s" % mexpr
+        )
+    return parts[0], parts[1]
+
+
+# performance optimize
+# 0b01 for request
+# 0b10 for response
+mapchoice_option_mapper = {
+    "request": 0b01,
+    "response": 0b10,
+}
+
+
+class MapEditor:
+    def __init__(self):
+        self.map_list = []
+        self.map_choice = 0b10
+
+    def load(self, loader):
+        loader.add_option(
+            "mapeditor", typing.Sequence[str], [],
```

Comment:
This should describe what it is doing, not how we name that component internally. How about `map_to_file` (or `map_local`)?